src/inference/src/ie_core.cpp

```diff
@@ -245,32 +245,74 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             std::shared_ptr<ie::ICacheManager> _cacheManager;
         };
 
-        bool flag_allow_auto_batching = true;
+        // Core default supported properties and values, which are core global properties.
+        // They don't require a specified device.
+        const ov::AnyMap default_core_supported_properties = {ov::cache_dir(""),
+                                                              ov::hint::allow_auto_batching(true),
+                                                              ov::auto_batch_timeout(1000),
+                                                              ov::force_tbb_terminate(false)};
+
+        void set_core_config(ov::AnyMap& config) {
+            for (auto it = config.begin(); it != config.end();) {
+                auto item = default_core_supported_properties.find(it->first);
+                if (item == default_core_supported_properties.end()) {
+                    it++;
+                    continue;
+                }
 
-        void setAndUpdate(ov::AnyMap& config) {
-            auto it = config.find(CONFIG_KEY(CACHE_DIR));
-            if (it != config.end()) {
-                std::lock_guard<std::mutex> lock(_cacheConfigMutex);
-                fillConfig(_cacheConfig, it->second.as<std::string>());
-                for (auto& deviceCfg : _cacheConfigPerDevice) {
-                    fillConfig(deviceCfg.second, it->second.as<std::string>());
+                if (it->first == ov::cache_dir.name()) {
+                    std::lock_guard<std::mutex> lock(_cacheConfigMutex);
+                    fillConfig(_cacheConfig, it->second.as<std::string>());
+                    for (auto& deviceCfg : _cacheConfigPerDevice) {
+                        fillConfig(deviceCfg.second, it->second.as<std::string>());
+                    }
+                } else if (it->first == ov::force_tbb_terminate.name()) {
+                    auto flag = it->second.as<std::string>() == CONFIG_VALUE(YES) ? true : false;
+                    executorManager()->setTbbFlag(flag);
+                }
+
+                {
+                    std::lock_guard<std::mutex> lock(_commonConfigMutex);
+                    auto temp = _commonConfig.find(it->first);
+                    if (temp != _commonConfig.end()) {
+                        _commonConfig[it->first] = it->second;
+                    } else {
+                        _commonConfig.insert(*it);
+                    }
+                }
+                it = config.erase(it);
```

Comment:
config will be changed, maybe config need to be read-only. or it++ is enough?

Reply:
This path is for core property without device name, the global config will be removed from this `config`.